client/my-sites/stats/hooks/use-should-gate-stats.ts

```diff
@@ -34,7 +34,7 @@ export const shouldGateStats = ( state: object, siteId: number | null, statType:
 	const siteHasPaidStats = siteHasFeature( state, siteId, FEATURE_STATS_PAID );
 
 	// check site type
-	if ( jetpackSite || atomicSite ) {
+	if ( jetpackSite && ! atomicSite ) {
```

Comment:
I think this will gate a self-hosted Jetpack site that decided to check their stats in calypso.